torchquad/integration/monte_carlo.py

```diff
@@ -1,64 +1,240 @@
 
         Raises:
             ValueError: If len(integration_domain) != dim
 
         Returns:
-            torch.Tensor: integral value
+            backend-specific number: Integral value
         """
         self._check_inputs(dim=dim, N=N, integration_domain=integration_domain)
-        logger.debug(
-            "Monte Carlo integrating a "
-            + str(dim)
-            + "-dimensional fn with "
-            + str(N)
-            + " points over "
-            + str(integration_domain),
+        logger.opt(lazy=True).debug(
+            "Monte Carlo integrating a {dim}-dimensional fn with {N} points over {dom}",
+            dim=lambda: dim,
+            N=lambda: N,
+            dom=lambda: integration_domain,
         )
+        integration_domain = _setup_integration_domain(dim, integration_domain, backend)
+        sample_points = self.calculate_sample_points(N, integration_domain, seed, rng)
+        logger.debug("Evaluating integrand")
+        function_values, self._nr_of_fevals = self.evaluate_integrand(fn, sample_points)
+        return self.calculate_result(function_values, integration_domain)
+
+    def get_jit_compiled_integrate(
+        self, dim, N=1000, integration_domain=None, seed=None, backend="torch"
+    ):
+        """Create an integrate function where the performance-relevant steps except the integrand evaluation are JIT compiled.
+        Use this method only if the integrand cannot be compiled.
+        The compilation happens when the function is executed the first time.
+        With PyTorch, return values of different integrands passed to the compiled function must all have the same format, e.g. precision.
+
+        Args:
+            dim (int): Dimensionality of the integration domain.
+            N (int, optional): Number of sample points to use for the integration. Defaults to 1000.
+            integration_domain (list or backend tensor, optional): Integration domain, e.g. [[-1,1],[0,1]]. Defaults to [-1,1]^dim. It also determines the numerical backend if possible.
+            seed (int, optional): Random number generation seed for the sequence of sampling point calculations, only set if provided. The returned integrate function calculates different points in each invocation with and without specified seed. Defaults to None.
+            backend (string, optional): Numerical backend. This argument is ignored if the backend can be inferred from integration_domain. Defaults to "torch".
+
+        Returns:
+            function(fn, integration_domain): JIT compiled integrate function where all parameters except the integrand and domain are fixed
+        """
+        self._check_inputs(dim=dim, N=N, integration_domain=integration_domain)
+        integration_domain = _setup_integration_domain(dim, integration_domain, backend)
+        backend = infer_backend(integration_domain)
+        # autoray's autojit function does JIT compilation, too, but in its
```

Comment:
I think my explanation about why I didn't use autojit is inaccurate, so I've changed it now.
autojit supports functions where all arguments are tensors, which is similar to torch.jit.trace and is supported by JAX and TensorFlow as well.
It could be possible to use autojit, but it's not in autoray 0.2.5 and differences between backends would probably require case distinctions anyway, for example the differences in how random numbers are generated.